tests/platform_tests/cli/test_show_platform_dpu.py

```diff
@@ -0,0 +1,183 @@
+from tests.common.platform.device_utils_dpu import *
+from tests.common.helpers.platform_api import chassis, module
+from tests.platform_tests.api.conftest import platform_api_conn
+from tests.common.devices.sonic import *
+
+pytestmark = [
+    pytest.mark.topology('t1')
+]
+
+def test_midplane_ip(duthosts, enum_rand_one_per_hwsku_hostname, platform_api_conn):
+    """
+    @summary: Verify `Midplane ip address between NPU and DPU`
+    """
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    num_modules = int(chassis.get_num_modules(platform_api_conn))
+    ping_count = 0
+    dpu_count = 0
+    ip_address_list = []
+
+    output_dpu_status = duthost.show_and_parse('show chassis module status')
+
+    for index in range(len(output_dpu_status)):
+        parse_output = output_dpu_status[index]
+        if 'DPU' in parse_output['name']:
+            if parse_output['oper-status'] != 'Offline':
+                index = (parse_output['name'])[-1]
+                # Api will always return the ip address as it is statically configured now
+                # Check will be added for ip address once dynamic implementation is in place
+                ip_address_list.append(module.get_midplane_ip(platform_api_conn, index))
+
+    check_dpu_ping_status(duthost, ip_address_list)
+
+def test_link_flap(duthosts, enum_rand_one_per_hwsku_hostname):
+    """
+    @summary: Verify `Link flap between NPU and DPU`
+    - Bringing all DPU interfaces down.
+    - Deleting the state table from redis db
+    - Checking show system-health dpu to see no output is there.
+      (Since the db entries are deleted and link down, state table would be empty which gets reflected in the cli)
+    - Brining up the interfaces.
+    - State table gets updated automatically after a minute
+    - Checking the show system-health dpu cli to get all the details
+    """
+
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    count_up = 0
+    count_down = 0
+    dpu_db_before = 0
+    dpu_db_after = 0
+
+    # Checking the state of dpu health before the flap
+    dpu_db_before = count_dpu_modules_in_system_health_cli(duthost)
+
+    output_interface_cmd = duthost.show_and_parse('show ip interface')
+    for index in range(len(output_interface_cmd)):
+        parse_output = output_interface_cmd[index]
+        interface = parse_output['interface']
+
+        if interface != "eth0" and 'eth' in interface:
+            output_intf_down = duthost.shell("ifconfig %s down"%(interface))["stdout_lines"]
+            time.sleep(1)
+
+    # Deleting the DB table entry after bringing down the interface
+    duthost.shell(
+            "for key in `redis-cli -p 6380 -h 127.0.0.1 -n 13 --scan --pattern \"DPU_STATE|DPU*\" `; "
+            "do redis-cli -p 6380 -h 127.0.0.1 -n 13 del $key ; done")
+
+    # Checking the state of dpu health again after bringing down interface and deleting db entry
+    dpu_db = wait_until(60, 60, 0, count_dpu_modules_in_system_health_cli, duthost)
```

Comment:
Looks like here the expectation is the failure of the wait_until, which actually equals to a 60s sleep, isn't it? And if in the first check the return value of  count_dpu_modules_in_system_health_cli is not 0, the wait_until will pass and the following pytest_assert will fail?

Reply:
Looks like here the expectation is the failure of the wait_until, which actually equals to a 60s sleep, isn't it?  Yes

And if in the first check the return value of count_dpu_modules_in_system_health_cli is not 0, the wait_until will pass and the following pytest_assert will fail?  Yes